OpenRA.Mods.RA/Widgets/ProductionPaletteWidget.cs

```diff
@@ -87,6 +90,35 @@ public ProductionPaletteWidget(OrderManager orderManager, World world, WorldRend
 			clock = new Animation(world, "clock");
 		}
 
+		public void ScrollDown()
+		{
+			var totalRows = AllBuildables.Count() / Columns;
```

Comment:
You can skip the check below by changing this to `var totalRows = (AllBuildables.Count() + Columns - 1) / Columns;`